modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ElixirClientCodegen.java

```diff
@@ -613,7 +613,7 @@ public String decodedStruct() {
                 return "%{}";
             }
             // Primitive return type, don't even try to decode
-            if (baseType == null || (simpleType && primitiveType)) {
+            if (baseType == null || (containerType == null && primitiveType)) {
```

Comment:
@niku FYI. I replaced `simpleType` with containerType` in Elixir client generator and there's no change in the output.